common/network-shuffle/src/main/java/org/apache/spark/network/shuffle/RemoteBlockPushResolver.java

```diff
@@ -1933,6 +1933,39 @@ public RoaringBitmap getMapTracker() {
     int getNumIOExceptions() {
       return numIOExceptions;
     }
+
+    private record ResourceCleaner(
+        File dataFile,
+        FileChannel dataChannel,
+        MergeShuffleFile indexFile,
+        MergeShuffleFile metaFile,
+        AppAttemptShuffleMergeId appAttemptShuffleMergeId,
+        int reduceId) implements Runnable {
+
+      @Override
+      public void run() {
+        try {
```

Comment:
My only question is that this introduces some duplicate code with the `AppShufflePartitionInfo#closeAllFilesAndDeleteIfNeeded` method. Is there really no way to eliminate this duplication?

Reply:
+1 reuse `AppShufflePartitionInfo#closeAllFilesAndDeleteIfNeeded`.